### PR TITLE
TF-2737 Separate PDF view handler between each browser

### PR DESCRIPTION
**Pull request by tddang-linagora**
### Issue
- #2737 

### Demo

**Chrome**

https://github.com/linagora/tmail-flutter/assets/160106668/d5ca3a56-b648-4b12-884d-bcc34265208c

**Firefox**

https://github.com/linagora/tmail-flutter/assets/160106668/fc124e4d-7e87-48ea-9bfc-3a9f3c7a6bb1

**Safari**

https://github.com/linagora/tmail-flutter/assets/160106668/99d7e621-65f3-4b04-b2bc-31453f4f99b1

### Long PDF name screenshot
Screenshot 2024-03-22 at 16 08 30

### CODE REVIEW

File: core/lib/data/network/download/download_manager.dart [hunk 1]
```diff
@@ -109,17 +109,24 @@ class DownloadManager {
     }
   }
 
-  Future<void> openDownloadedFileWeb(Uint8List bytes, String? mimeType) async {
+  Future<void> openDownloadedFileWeb(
+    Uint8List bytes,
+    String? mimeType,
+    String? fileName
```
Comment (hoangdat) [inline core/lib/data/network/download/download_manager.dart:115]: what happen if `fileName` is null?

File: core/lib/utils/html/html_utils.dart [hunk 2]
```diff
@@ -151,44 +153,239 @@ class HtmlUtils {
...
+
+  static String get _pdfFileInfoStyle => '''
+    display: flex;
+    align-items: center;
+    padding: 5px 10px;''';
+
+  static String get _pdfAppbarElement => '''
+    <div id="app-bar">
+      <div id="file-info">
+        <span id="file-name" style="margin-right: 10px;"></span>
```
Comment (hoangdat) [inline core/lib/utils/html/html_utils.dart:344]: what happen if `fileName` is very long?

File: core/lib/utils/html/html_utils.dart [hunk 3]
```diff
@@ -151,44 +153,239 @@ class HtmlUtils {
...
+    z-index: 100; /* Ensure buttons stay on top */''';
+
+  static String get _pdfDownloadButtonStyle => '''
+    padding: 5px 10px;
+    border: 1px solid #ddd;
+    border-radius: 5px;
+    cursor: pointer;
+    margin-left: 10px;''';
+
+  static String get _pdfFileInfoStyle => '''
```
Comment (dab246) [inline core/lib/utils/html/html_utils.dart:336]: Why not use it as constant?

File: core/lib/utils/html/html_utils.dart (hunk 2 above)
Comment (tddang-linagora) [reply core/lib/utils/html/html_utils.dart:344]: added screenshot

File: core/lib/utils/html/html_utils.dart (hunk 3 above)
Comment (tddang-linagora) [reply core/lib/utils/html/html_utils.dart]: Done

File: core/lib/utils/html/html_utils.dart (hunk 2 above)
Comment (hoangdat) [reply core/lib/utils/html/html_utils.dart:344]: why not constraint it?

**Review by dab246 — APPROVED** [review]
lgtm

File: core/lib/utils/html/html_utils.dart (hunk 2 above)
Comment (tddang-linagora) [reply core/lib/utils/html/html_utils.dart:361]: Done

**Review by hoangdat — APPROVED** [review]

**Pull request merged**